### DISCUSSION TITLE
How can I use plain webserver as local package mirror

### DISCUSSION THREAD

**Discussion by wechmarer** [microsoft/vcpkg#27230] [Ideas]
How can I use a simple mirror for the vcpkg ports?

1. I want to use a local http/https-server as mirror. 
2. I want to copy the packages to my server by my own. vcpkg should not be responsible for that.
3. I want an option to set the behaviour in case a required package is not in the mirror

This is similary to the mirror concept of yocto / bitbake.

**Comment by autoantwort** [comment]
> I want to copy the packages to my server by my own. vcpkg should not be responsible for that.

What speaks against the http binary source? 

> I want an option to set the behaviour in case a required package is not in the mirror

Use `--only-binarycaching`

**Reply by wechmarer** [reply]
> What speaks against the http binary source?

Mirror the binaries inhibits any further code manipulation,(e.g. applying a patch) or usage of a new toolchain. This is more a closed source approach than an open source approach. Then vcpkg can skip the source code download and should provide prebuild binaries from a central server. This would save a lot of compile time at developers side. This would be the day vcpkg is skipped at all.

> Use `--only-binarycaching`

See above
